File: 2195-time-needed-to-buy-tickets/2195-time-needed-to-buy-tickets.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int time = 0;
        if(tickets[k] == 1) {
            return k + 1;
        }
        for(int i = 0; i < tickets.length; i++) {
            if(i <= k) {
                time += Math.min(tickets[i], tickets[k]);
            } else {
                time += Math.max(1, Math.min(tickets[i], tickets[k] - 1));
            }
        }
        return time;
    }
}

File: 2195-time-needed-to-buy-tickets/2195-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        time = 0
        if tickets[k] == 1:
            return k + 1
        for i in range(len(tickets)):
            if i <= k:
                time += min(tickets[i], tickets[k])
            else:
                time += max(1, min(tickets[i], tickets[k] - 1))
        return time